pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authorization/AuthorizationProvider.java

```diff
@@ -489,4 +490,36 @@ default Boolean allowTopicOperation(TopicName topicName,
             throw new RestException(e.getCause());
         }
     }
+
+    /**
+     * Check if a given <tt>role</tt> is allowed to execute a given topic <tt>operation</tt> on topic's <tt>policy</tt>.
+     *
+     * @param topic topic name
+     * @param role role name
+     * @param operation topic operation
+     * @param authData authenticated data
+     * @return CompletableFuture<Boolean>
+     */
+    default CompletableFuture<Boolean> allowTopicPolicyOperationAsync(TopicName topic,
+                                                                      String role,
+                                                                      PolicyName policy,
+                                                                      PolicyOperation operation,
+                                                                      AuthenticationDataSource authData) {
+        return FutureUtil.failedFuture(
```

Comment:
Can you make the default implementation to use the existing method? If people wrote its own plugin, it will fail all the topic policy operations when it upgrades to a newer version of Pulsar.

Reply:
So call `return isTenantAdmin(topic.getTenant(), role, tenantInfo , authData);`, correct? How to get tenantInfo from here?

Reply:
@sijie

Reply:
@zymap is working on some authorization changes. @zymap Can you take a look at this pull request?

Reply:
Overall looks good. This PR looks like only add the validation for the topic level authorization.

Reply:
It does.